include/oneapi/tbb/detail/_concurrent_skip_list.h

```diff
@@ -1089,11 +1094,15 @@ class concurrent_skip_list {
     }
 
     void delete_node( node_ptr node ) {
-        size_type sz = calc_node_size(node->height());
-
-        // Destroy the node
+        //Destroy the level pointers
+        for (size_type l = 0; l < node->height(); ++l) {
+            node_allocator_traits::destroy(my_node_allocator, &node->atomic_next(l));
+        }
+        // Destroy the node itself
         node_allocator_traits::destroy(my_node_allocator, node);
+
         // Deallocate the node
+        size_type sz = calc_node_size(node->height());
```

Comment:
Seems like UB takes place here. `sz` calculation executes `node->height()`, but the destructor of `node` was called on the previous line.

Reply:
good catch